downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -263,9 +263,9 @@ abstract class VersionControlSystem {
     abstract fun getWorkingTree(vcsDirectory: File): WorkingTree
 
     /**
-     * Return true if any of the [aliases] matches [vcsType]. Comparison is done case-insensitively.
+     * Return true if this VCS can handle the named [vcsType].
      */
-    fun isApplicableType(vcsType: String) = vcsType.toLowerCase() in aliases
+    fun isApplicableType(vcsType: String) = (aliases + type).any { it.equals(vcsType, true) }
```

Comment:
This changes the case of the type entries, e.g. from "gitrepo" to "GitRepo". Did you verify that this does not cause issues?

Reply:
I don't think the type entries change, unless I'm overlooking something. Note that I haven't changed the value `type` is assigned to. If you mean that `(aliases + type)` now contains (probably) lowercase aliases and the camel-case `type` name whereas the list before only contained lower-case aliases including the lower-cased type, that should not cause any problems as the comparison was already (half) case-insensitive before, and `isApplicableType()` is the only place where `aliases` is used. I say "half case-insensitive" because before only the `vcsType` was lower-cased and aliases where by convention, whereas now aliases could also have a different casing.